crates/unftp-sbe-fs/src/lib.rs

```diff
@@ -123,35 +120,30 @@ impl<User: UserDetail> StorageBackend<User> for Filesystem {
         P: AsRef<Path> + Send + Debug,
         <Self as StorageBackend<User>>::Metadata: Metadata,
     {
-        let full_path: PathBuf = self.full_path(path).await?;
-
-        let prefix: PathBuf = self.root.clone();
-
-        let mut rd: tokio::fs::ReadDir = tokio::fs::read_dir(full_path).await?;
-
-        let mut fis: Vec<Fileinfo<std::path::PathBuf, Self::Metadata>> = vec![];
-        while let Ok(Some(dir_entry)) = rd.next_entry().await {
-            let prefix = prefix.clone();
-            let path = dir_entry.path();
-            let relpath = path.strip_prefix(prefix).unwrap();
-            let relpath: PathBuf = std::path::PathBuf::from(relpath);
-            let metadata = tokio::fs::symlink_metadata(dir_entry.path()).await?;
-            let meta: Self::Metadata = Meta { inner: metadata };
-            fis.push(Fileinfo { path: relpath, metadata: meta })
-        }
+        let path = strip_prefixes(path.as_ref());
+
+        let fis: Vec<Fileinfo<std::path::PathBuf, Self::Metadata>> = cap_fs::read_dir(self.root_fd.clone(), path)
+            .and_then(|dirent| {
+                let entry_path: PathBuf = dirent.file_name().into();
+                cap_fs::symlink_metadata(self.root_fd.clone(), path.join(entry_path.clone())).map_ok(move |meta| {
+                    let metadata = Meta { inner: meta };
+                    Fileinfo { path: entry_path, metadata }
+                })
+            })
+            .try_collect::<Vec<_>>()
+            .await?;
 
         Ok(fis)
     }
 
     //#[tracing_attributes::instrument]
     async fn get<P: AsRef<Path> + Send + Debug>(&self, _user: &User, path: P, start_pos: u64) -> Result<Box<dyn tokio::io::AsyncRead + Send + Sync + Unpin>> {
-        use tokio::io::AsyncSeekExt;
-
-        let full_path = self.full_path(path).await?;
-        let mut file = tokio::fs::File::open(full_path).await?;
+        let path = path.as_ref().strip_prefix("/").unwrap_or(path.as_ref());
+        let mut file = cap_fs::open(self.root_fd.clone(), path).await?;
         if start_pos > 0 {
-            file.seek(std::io::SeekFrom::Start(start_pos)).await?;
+            file.seek(std::io::SeekFrom::Start(start_pos))?;
```

Comment:
This is a blocking call in async code not so?

Reply:
Yes, but only in the most technically pedantic way.  Tokio considers all file system operations to be blocking.  But seek doesn't actually do any I/O at all, even for network file systems like NFS or FUSE.  It just updates the file pointer in-kernel.  There's nothing that could block.  Still, I could change it if you want.  Do you want me to?

Reply:
OK, no thank you for the explanation, perhaps add this explanation as a comment for future readers.

Reply:
Actually, I went ahead and "fixed" it anyway, since I was doing the same thing on line 164.